connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -232,6 +233,9 @@ public class DistributedHerder extends AbstractHerder implements Runnable {
 
     private final DistributedConfig config;
 
+    // visible for testing
+    public Future<?> herderFuture;
```

Comment:
This field should still be `private`, and only exposed via a getter:
```java
private Future<?> herderFuture;

...

public Future<?> herderFuture() {
    return herderFuture;
}
```

Otherwise, it could be modified by other classes, which we almost certainly do not want.